File: pset6/readability/readability.py
# Get input text
text = input("Text: ")

letters = 0
words = 0
sentences = 0

# Count letters
for char in text:
    if char.isalpha():
        letters += 1

# Count words
words = len(text.split())

# Count sentences
for char in text:
    if char in ['.', '!', '?']:
        sentences += 1

# Calculate L and S
L = (letters / words) * 100
S = (sentences / words) * 100

# Coleman-Liau index
index = round(0.0588 * L - 0.296 * S - 15.8)

# Output result
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Ask user for text
    string text = get_string("Text: ");

    int letters = 0;
    int words = 1;     // start from 1 because last word has no space after it
    int sentences = 0;

    // Count letters, words, sentences
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    // Calculate L and S
    float L = (letters * 100.0) / words;
    float S = (sentences * 100.0) / words;

    // Coleman-Liau formula
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Print result
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}
